pkg/controllers/allocation/scheduling/scheduler.go

```diff
@@ -74,10 +75,10 @@ func (s *Scheduler) Solve(ctx context.Context, provisioner *v1alpha5.Provisioner
 	defer metrics.Measure(schedulingDuration.WithLabelValues(provisioner.Name))()
 
 	constraints := provisioner.Spec.Constraints.DeepCopy()
+	constraints.Labels = functional.UnionStringMaps(constraints.Labels, map[string]string{v1alpha5.ProvisionerNameLabelKey: provisioner.Name})
```

Comment:
Do we need to store this back to `constraints.Labels` can we directly pass them to the `With()` on line 81?

Reply:
Yes -- unfortunately this a little bit convoluted. The labels are propagated to nodes, but the requirements aren't. However, we include the labels in the requirements when doing scheduling calculation so that we can ensure pod requirements are met.

Reply:
ok I see, what you mean we are using the constraints object later.